src/main/java/com/exactpro/th2/readlog/RegexLogParser.java

```diff
@@ -40,47 +40,53 @@ public RegexLogParser(Map<String, AliasConfiguration> cfg) {
     }
 
     public LogData parse(String alias, String raw) {
-        LogData resultData = new LogData();
 
         AliasConfiguration configuration = cfg.get(alias);
         if (configuration == null) {
             logger.error("Unknown alias {}, there no configuration", alias);
-            return new LogData();
+            throw new IllegalArgumentException("Unknown alias '" + alias +"'. No configuration found" );
         }
 
+        LogData resultData = new LogData();
         List<Integer> regexGroups = configuration.getGroups();
         if (regexGroups.isEmpty()) {
             parseBody(raw, configuration.getRegexp(), resultData);
         } else {
             parseBody(raw, configuration.getRegexp(), regexGroups, resultData);
         }
 
+        if (resultData.getBody().isEmpty()) {
+            // fast way, nothing matches the regexp so we don't need to check for date pattern
+            return resultData;
+        }
+
         // Timestamp string from log
         Pattern datePattern = configuration.getTimestampRegexp();
-        if (datePattern!=null) {
-            if (!lookForTimestamp(raw, datePattern, resultData)) return new LogData();
+        if (datePattern != null) {
+            if (!lookForTimestamp(raw, datePattern, resultData)) {
+                throw new IllegalStateException("The pattern '" + datePattern.pattern() + "' cannot extract the timestamp from the string: " + raw);
+            }
         }
 
         // DateTime from log
         String timestampFormat = configuration.getTimestampFormat();
         if (timestampFormat != null && !timestampFormat.isEmpty()) {
-            if (!parseTimestamp(timestampFormat, resultData)) return new LogData();
+            parseTimestamp(timestampFormat, resultData);
         }
 
         return resultData;
     }
 
-    private boolean parseTimestamp(String format, LogData data) {
+    private void parseTimestamp(String format, LogData data) {
+        String rawTimestamp = data.getRawTimestamp();
         try {
             DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
```

Comment:
Maybe cache this value to reuse in the future

Reply:
I have moved the formatter creation to the configuration. So, the formatted will be created only once